src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts

```diff
@@ -35,12 +35,161 @@ class F extends GPUTest {
+      (dstBlocksPerRow * dstBlockRowsPerImage - 1) * bytesPerDstAlignedBlockRow +
+      align(dstBlocksPerRow * bytesPerBlock, 4);
+    const dstBufferDesc: GPUBufferDescriptor = {
+      size: dstBufferSize,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    };
+    const dstBuffer = this.device.createBuffer(dstBufferDesc);
+
+    encoder.copyTextureToBuffer(
+      { texture: dstTexture, mipLevel: dstCopyLevel },
+      {
+        buffer: dstBuffer,
+        bytesPerRow: bytesPerDstAlignedBlockRow,
+        rowsPerImage: dstTextureSizeAtLevel.height,
+      },
+      dstTextureSizeAtLevel
+    );
+    this.device.defaultQueue.submit([encoder.finish()]);
+
+    // Fill expectedDataWithPadding with the expected data of dstTexture. The other values in
+    // expectedDataWithPadding are kept 0 to check if the texels untouched by the copy are 0
+    // (their previous values).
+    const expectedDataWithPadding = new ArrayBuffer(dstBufferSize);
+    const expectedUint8DataWithPadding = new Uint8Array(expectedDataWithPadding);
+    const expectedUint8Data = new Uint8Array(initialSrcData);
+
+    const appliedCopyBlocksPerRow = appliedCopyWidth / blockWidth;
+    const appliedCopyBlockRowsPerImage = appliedCopyHeight / blockHeight;
+    const srcCopyOffsetInBlocks = {
+      x: copyBoxOffsets.srcOffset.x / blockWidth,
+      y: copyBoxOffsets.srcOffset.y / blockHeight,
+      z: copyBoxOffsets.srcOffset.z,
+    };
+    const dstCopyOffsetInBlocks = {
+      x: copyBoxOffsets.dstOffset.x / blockWidth,
+      y: copyBoxOffsets.dstOffset.y / blockHeight,
+      z: copyBoxOffsets.dstOffset.z,
+    };
+
+    for (let z = 0; z < appliedCopyDepth; ++z) {
+      const srcOffsetZ = srcCopyOffsetInBlocks.z + z;
+      const dstOffsetZ = dstCopyOffsetInBlocks.z + z;
+      for (let y = 0; y < appliedCopyBlockRowsPerImage; ++y) {
+        const dstOffsetYInBlocks = dstCopyOffsetInBlocks.y + y;
+        const expectedDataWithPaddingOffset =
+          bytesPerDstAlignedBlockRow * (dstBlockRowsPerImage * dstOffsetZ + dstOffsetYInBlocks) +
+          dstCopyOffsetInBlocks.x * bytesPerBlock;
+
+        const srcOffsetYInBlocks = srcCopyOffsetInBlocks.y + y;
+        const expectedDataOffset =
+          bytesPerBlock *
+            srcBlocksPerRow *
+            (srcBlockRowsPerImage * srcOffsetZ + srcOffsetYInBlocks) +
+          srcCopyOffsetInBlocks.x * bytesPerBlock;
+
+        expectedUint8DataWithPadding.set(
+          expectedUint8Data.slice(
+            expectedDataOffset,
+            expectedDataOffset + appliedCopyBlocksPerRow * bytesPerBlock
+          ),
+          expectedDataWithPaddingOffset
+        );
+      }
+    }
+
+    // Verify the content of the whole subresouce of dstTexture at dstCopyLevel (in dstBuffer) is expected.
+    this.expectContents(dstBuffer, expectedUint8DataWithPadding);
+  }
 }
 
 export const g = makeTestGroup(F);
 
 // TODO(jiawei.shao@intel.com): support all WebGPU texture formats
```

Comment:
nit (new guidance): move this TODO inside the description so it's visible in /standalone/. Also make it just a little more descriptive.